src/toil/jobStores/aws/jobStore.py

```diff
@@ -726,6 +810,70 @@ def reader():
                 thread.join()
                 assert bool(self.version) == (self.content is None)
 
+        def copyFrom(self, srcKey):
+            """
+            Copies contents of source key into file.
+
+            :param srcKey: The key that will be copied from
+            """
+            size = srcKey.bucket.get_key(srcKey.name).size
+            if size <= self._maxInlinedSize():
+                self.content = srcKey.get_contents_as_string()
+            else:
+                self.version = self._copyKey(newKeyName=self._fileID,
+                                             dstBucketName=self.outer.filesBucket.name,
+                                             srcBucketName=srcKey.bucket.name,
+                                             srcKey=srcKey,
+                                             headers=self._s3EncryptionHeaders()).version_id
+
+        def copyTo(self, newKeyName, dstBucket):
+            """
+            Copies contents of file to destination key.
+
+            :param str newKeyName: The name of the new key to be created or copied to from the file.
+            :param dstBucket: The bucket that contains or will contain the destination key.
```

Comment:
`from boto.s3.bucket import Bucket` at the top and `:param Bucket dstBucket:` here.